### PR TITLE
OSSM-1698 Restrict istio-cni-node DaemonSet

### PR DESCRIPTION
**Pull request by InnocentK**
`istio-cni` by default is installed on all nodes. This PR adds the ability to exclude `istio-cni` from being installed on specified nodes by adding the `maistra.io/exclude-cni `label to that node and setting it to true which updates the `istio-cni` `nodeSelector` daemon set.

Replaces https://github.com/maistra/istio/pull/675 as istio-operator's istio-cni helm chart overwrites the one from istio.

### CODE REVIEW

**Review by jewertow — COMMENTED** [review]
You have to apply these changes to the corresponding file in the directory resources/helm/overlays and then run "make gen".

**Comment by luksa** [comment]
@InnocentK To expand on @jewertow's comment: you actually need to make the same change in two files: 
- `resources/helm/overlays/istio_cni/templates/daemonset.yaml`
- `resources/helm/overlays/istio_cni/templates/daemonset-v2.3.yaml`

Then run `make gen` to update the files in `resources/v2.3/`

File: manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ metadata:
       The Maistra Operator enables you to install, configure, and manage an instance of Maistra service mesh. Maistra is based on the open source Istio project.
 
     containerImage: quay.io/maistra/istio-ubi8-operator:2.3.0
-    createdAt: 2022-10-14T13:24:19EDT
+    createdAt: 2022-11-09T15:48:27EST
```
Comment (luksa) [inline manifests-maistra/2.3.0/maistraoperator.v2.3.0.clusterserviceversion.yaml]: Please revert this change. Make gen always updates the timestamp, but we usually revert it when this is the only change in the file.

File: manifests-servicemesh/2.3.0/servicemeshoperator.v2.3.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -17,7 +17,7 @@ metadata:
       The OpenShift Service Mesh Operator enables you to install, configure, and manage an instance of Red Hat OpenShift Service Mesh. OpenShift Service Mesh is based on the open source Istio project.
 
     containerImage: ${OSSM_OPERATOR_IMAGE}
-    createdAt: 2022-10-14T13:24:21EDT
+    createdAt: 2022-11-09T15:48:28EST
```
Comment (luksa) [inline manifests-servicemesh/2.3.0/servicemeshoperator.v2.3.0.clusterserviceversion.yaml]: Same here.

**Comment by InnocentK** [comment]
Thanks for the help @luksa @jewertow. I made the changes you both suggested

**Review by luksa — APPROVED** [review]
LGTM. Works as expected:
```
$ k label node master-1 maistra.io/exclude-cni=true
node/master-1 labeled

$ kgno -L maistra.io/exclude-cni
NAME       READY    ROLES    AGE   VERSION           EXCLUDE-CNI
master-1   Ready    master   91m   v1.20.0+87cc9a4   true
master-2   Ready    master   91m   v1.20.0+87cc9a4
master-3   Ready    master   91m   v1.20.0+87cc9a4
worker-1   Ready    worker   84m   v1.20.0+87cc9a4
worker-2   Ready    worker   84m   v1.20.0+87cc9a4
worker-3   Ready    worker   83m   v1.20.0+87cc9a4

$ kgpo
NAME                              READY   STATUS        RESTARTS   AGE
istio-cni-node-v2-3-8gw6m         1/1     Running       0          83s
istio-cni-node-v2-3-gfhs7         1/1     Running       0          83s
istio-cni-node-v2-3-h4ccr         0/1     Terminating   0          83s
istio-cni-node-v2-3-llhzh         1/1     Running       0          83s
istio-cni-node-v2-3-mntvh         1/1     Running       0          83s
istio-cni-node-v2-3-zng79         1/1     Running       0          83s
istio-operator-686c7669f8-zr5vt   1/1     Running       1          4m24s
```

The v2.2- pods also behave correctly.

**Comment by jwendell** [comment]
I wonder why we didn't implement this in upstream first.

**Comment by InnocentK** [comment]
Closed in favor of moving these changes to upstream (https://github.com/istio/istio/pull/42038).
